### DISCUSSION TITLE
'New discussion' in current category

### DISCUSSION THREAD

**Discussion by sudojunior** [community/community#2899] [Discussions] [answered]
Based on the premise of selecting a category to narrow the search results, would it be plausible and / or ideal to apply the same logic to the **New discussion** action?

There are some caveats with this approach where certain fields may be filled in by default for templates (ref #2838 which would mirror the current flow of issue creation), and some others that are yet to be mentioned here.

> `/github/feedback/discussions/new?category=discussions-feedback` as an example for _this repository category_.

**Comment by evi-liu** [comment, marked as answer]
Thanks for the feedback! I'm not sure I completely follow. To clarify, is this request to have a filter on new discussions? Or something else?

**Reply by sudojunior** [reply]
> ..., is this request to have a filter on new discussions? Or something else?

It would be intended to auto resolve the current category _if one were selected_, like how issues might work (but without the choosing view - at `/issues/new/choose`).

[github/docs: new issue with `improve-the-site.md` template](https://github.com/github/docs/issues/new?assignees=&labels=engineering&template=improve-the-site.md&title=)

EDIT: In all repository settings pages, there is an action to setup a first discussion for that repository, it uses a `~/discussions/new?category=general&welcome_text=true` query to guide the end-user towards making their first post. However, I have been unable to get it working when whitespace is involved.

**Reply by sudojunior** [reply]
This feature appears to be implemented or [shipped](https://github.com/github/feedback/discussions/categories/discussions-feedback?discussions_q=category%3A"Discussions+Feedback"+label%3A"🚀+shipped") as per the label. _When attempting to track its history, this appears to have been implemented some time around September._

And so with that, I mark this as the answer - [check below as well](https://github.com/github/feedback/discussions/2899#discussioncomment-187809) query parameters for issues are also compatible with discussions (again unsure when that may have happened).

**Comment by w00fz** [comment]
Yes please! Query for pre-filling a discussions would be extremely useful to help getting a Discussion formatted in a certain desired way.

Please note that this is already possible with **Issues**, here is an example:

https://github.com/github/testrepo/issues/new?title=My%20Test%20Issue&body=**Here%20goes%20my%20body**%0A%0AWould%20be%20very%20useful%20in%20Discussions%20too.

The same could be for Discussions:

```
https://github.com/repo/owner/discussions/new?
  title=<discussion_title>&
  body=<discussion_body>&
  category=<discussion_category>
```

**Comment by sudojunior** [comment]
I'm aware that this is well and truly stale, but I went ahead and made it as an extension *if you can call it that*.
**This will only modify the 'New discussion' action as described in the title.**

https://github.com/sudojunior/discuss-in-category-github *Archived, but remains open.*

Only runs if the path matches to `github.com/:owner/:repo/discussions{/categories/:category}`.

_This will not be published on any extension store, nor will it be actively updated._

> If this is considered as self-promotion / off-topic, please remove / edit as necessary.
